qiskit/algorithms/linear_solvers/hhl.py

```diff
@@ -0,0 +1,373 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The HHL algorithm."""
+
+from typing import Optional, Union, List, Callable, Tuple
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister, AncillaRegister
+from qiskit.circuit.library import PhaseEstimation
+from qiskit.circuit.library.arithmetic.piecewise_chebyshev import PiecewiseChebyshev
+from qiskit.circuit.library.arithmetic.exact_reciprocal import ExactReciprocal
+from qiskit.opflow import Z, I, StateFn, TensoredOp, ExpectationBase, CircuitSampler, ListOp
+from qiskit.providers import Backend
+from qiskit.quantum_info import Operator
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.utils import QuantumInstance
+
+from .linear_solver import LinearSolver, LinearSolverResult
+from .matrices.linear_system_matrix import LinearSystemMatrix
+from .observables.linear_system_observable import LinearSystemObservable
+
+
+class HHL(LinearSolver):
+    """The HHL algorithm to solve systems of linear equations"""
+
+    def __init__(self,
+                 epsilon: float = 1e-2,
+                 expectation: Optional[ExpectationBase] = None,
+                 quantum_instance: Optional[Union[Backend, QuantumInstance]] = None) -> None:
```

Comment:
These 2 params do not seem to have any docstring. Not sure why the build here does not catch that.

Reply:
added docstrings